### PR TITLE
Center of mass computation

### PR DESCRIPTION
**Pull request by bzinberg**
# Summary
Adds a method `centerofmass(p::Polyhedron)` which returns the center of mass of `p`.

## Tested
Added unit tests which pass.  (Other existing unit tests appear to be broken, as seen in CI.)

### CODE REVIEW

File: docs/src/utilities.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ polar
 ```@docs
 volume
 surface
+centerofmass
```
Comment (blegat) [inline docs/src/utilities.md:23]: I'd rather have `center_of_mass`

File: src/polyhedron.jl [hunk 2]
```diff
@@ -61,6 +61,28 @@ function volume_simplex(s)
...
+Throws an error if `p` is degenerate.
+"""
+function centerofmass(p::Polyhedron{T}) where T
+    simplices = triangulation(p)
+    isempty(simplices) && error("Tried to compute center of mass of a degenerate polyhedron")
+    unscaled_center = zeros(T, fulldim(p))
+    unscaled_vol = zero(T)
+    for Δ in simplices
+        v = unscaled_volume_simplex(Δ)
+        # Note, the scaling constants for `unscaled_center` and `unscaled_vol`
```
Comment (blegat) [inline src/polyhedron.jl:77]: I don't understand this comment

File: src/polyhedron.jl [hunk 3]
```diff
@@ -61,6 +61,28 @@ function volume_simplex(s)
...
+    centerofmass(p::Polyhedron{T}) where {T}
+
+Returns the center of mass of `p`, represented as a `Vector{T}` of length `fulldim(p)`.
+Throws an error if `p` is degenerate.
+"""
+function centerofmass(p::Polyhedron{T}) where T
+    simplices = triangulation(p)
+    isempty(simplices) && error("Tried to compute center of mass of a degenerate polyhedron")
+    unscaled_center = zeros(T, fulldim(p))
+    unscaled_vol = zero(T)
```
Comment (blegat) [inline src/polyhedron.jl:74]: Add a comment explaining the approach here: e.g. the centroid is easy to compute (it's the average of the coordinates of the vertices and the centroid and center of mass coincides for simplices so we compute the center of mass of all simplicies to reduce the problem to the center of mass of finitely many objects for which we can just take the average weighted by the volume.

**Comment by blegat** [comment]
Thanks for the contribution, it would indeed be a nice addition of Polyhedra. I just have a few minor comments

**Comment by blegat** [comment]
The tests should be passing now if you rebase this PR with master.

File: src/polyhedron.jl (hunk 3 above)
Comment (bzinberg) [reply src/polyhedron.jl:78]: Added.

**Review by bzinberg — COMMENTED** [review]
Thanks @blegat!  Made the changes you suggested, it seems there is one more CI failure.

File: src/polyhedron.jl (hunk 2 above)
Comment (bzinberg) [reply src/polyhedron.jl]: Removed.

File: docs/src/utilities.md (hunk 1 above)
Comment (bzinberg) [reply docs/src/utilities.md]: Done.

**Review by blegat — APPROVED** [review]
Looks good, thanks for the contribution !

**Pull request merged**